File: automation-export/template/samples/python/hello_world.py
"""Simple Hello World application for testing Python automation."""

__version__ = "1.0.0"


def hello_world(name: str = "World") -> str:
    """
    Return a greeting message.

    Args:
        name: The name to greet (default: "World")

    Returns:
        A greeting message string
    """
    if not isinstance(name, str):
        raise TypeError("Name must be a string")
    if not name.strip():
        raise ValueError("Name cannot be empty")

    return f"Hello, {name}!"


def main() -> None:
    """Main entry point."""
    print(hello_world())
    print(hello_world("Python"))


if __name__ == "__main__":
    main()

File: automation-export/template/samples/node/hello-world.js
/**
 * Simple Hello World application for testing Node.js automation.
 */

/**
 * Return a greeting message.
 *
 * @param {string} [name='World'] - The name to greet
 * @returns {string} A greeting message string
 * @throws {TypeError} When name is not a string
 * @throws {Error} When name is empty
 */
function helloWorld(name = 'World') {
  if (typeof name !== 'string') {
    throw new TypeError('Name must be a string');
  }

  if (name.trim() === '') {
    throw new Error('Name cannot be empty');
  }

  return `Hello, ${name}!`;
}

/**
 * Main entry point.
 */
function main() {
  console.log(helloWorld());
  console.log(helloWorld('Node.js'));
}

// Run main if this module is executed directly
if (require.main === module) {
  main();
}

module.exports = { helloWorld, main };
